contrib_bots/lib/converter.py

```diff
@@ -0,0 +1,228 @@
+                'nano': -9,
+                'micro': -6,
+                'milli': -3,
+                'centi': -2,
+                'deci': -1,
+                'deca': 1,
+                'hecto': 2,
+                'kilo': 3,
+                'mega': 6,
+                'giga': 9,
+                'tera': 12,
+                'peta': 15,
+                'exa': 18}
+
+    aliases = {'c': 'celsius',
+               'cm^3': 'cubic-centimeter',
+               'ft^3': 'cubic-foot',
+               'm^3': 'cubic-meter',
+               'y^3': 'cubic-yard',
+               'f': 'fahrenheit',
+               'ft': 'foot',
+               'g': 'gram',
+               'ha': 'hectare',
+               'in': 'inch',
+               'k': 'kelvin',
+               'l': 'liter',
+               'm': 'meter',
+               'mi': 'mile',
+               'nmi': 'nautical-mile',
+               'oz': 'ounce',
+               'lb': 'pound',
+               'cm^2': 'square-centimeter',
+               'ft^2': 'square-foot',
+               'in^2': 'square-inch',
+               'km^2': 'square-kilometer',
+               'm^2': 'square-meter',
+               'mi^2': 'square-mile',
+               'y^2': 'square-yard',
+               't': 'ton',
+               'y': 'yard'}
+
+    def isfloat(self, value):
+        try:
+            float(value)
+            return True
+        except ValueError:
+            return False
+
+    def round_to(self, x, digits):
+        return round(x, digits-int(floor(log10(abs(x)))))
+
+class ConverterHandler(object):
+    '''
+    This plugin allows users to make conversions between various units,
+    e.g. Celsius to Fahrenheit, or Kilobytes to Gigabytes.
+    It looks for messages of the format
+    '@convert <number> <original-unit> <new-unit>.
+    The message '@convert help' posts a short description of how to use
+    the plugin, along with a list of all supported units.
+    '''
+
+    units = Units()
+
+    def usage(self):
+        return '''
+               This plugin allows users to make conversions between
+               various units, e.g. Celsius to Fahrenheit,
+               or Kilobytes to Gigabytes. It looks for messages of
+               the format '@convert <number> <original-unit> <new-unit>.
+               The message '@convert help' posts a short description of
+               how to use the plugin, along with a list of
+               all supported units.
+               '''
+
+    def triage_message(self, message, client):
+        return ('@convert' in message['content'])
+
+    def handle_message(self, message, client, state_handler):
+        content = message['content']
+
+        words = content.lower().split()
+        conversion_occ = [idx for idx, x in enumerate(copy.copy(words)) if x == "@convert"]
+        results = []
+
+        for idx in conversion_occ:
+            if (idx + 1) < len(words) and words[idx + 1] == 'help':
+                results.append('Converter usage:\n' +
```

Comment:
To reduce the size of the function, you can make a class variable global variable called `help_message` and do `results.append(help_message)`.

What will happen if I do this?
> @convert help @convert help